sdks/java/io/thrift/src/main/java/org/apache/beam/sdk/io/thrift/ThriftSchema.java

```diff
@@ -183,7 +188,12 @@ private Schema schemaFor(Class<?> targetClass) {
   private Schema.Field beamField(FieldMetaData fieldDescriptor) {
     try {
       final FieldType type = beamType(fieldDescriptor.valueMetaData);
-      return Schema.Field.nullable(fieldDescriptor.fieldName, type);
+      switch (fieldDescriptor.requirementType) {
+        case TFieldRequirementType.REQUIRED:
+          return Schema.Field.of(fieldDescriptor.fieldName, type);
+        default:
+          return Schema.Field.nullable(fieldDescriptor.fieldName, type);
+      }
```

Comment:
Fields used to always be nullable for thrift schemas and the table integration test requires non-nullable fields. So I just added what was actually missing, i.e. marking `required` as non-nullable in beam. `default` is still nullable, the same as `optional`, for safety reasons -- as it's defined as "opt-in, req-out".